objects/src/mock/chain.rs

```diff
@@ -55,3 +55,247 @@ pub fn mock_chain_data(consumed_notes: &mut [Note]) -> ChainMmr {
+        rng: R,
+
+        // Builder for new [AccountId]s of faucets.
+        account_id_builder: AccountIdBuilder<R>,
+
+        // Objects that have been created and commited to a block.
+        //
+        // These can be used to perform additional operations on a block.
+        #[cfg_attr(feature = "serde", serde(skip))]
+        objects: Objects<R>,
+
+        // Objects that have been created and are waiting for a block.
+        //
+        // These objects will become available once the block is sealed.
+        #[cfg_attr(feature = "serde", serde(skip))]
+        pending_objects: Objects<R>,
+    }
+
+    impl<R: Rng + SeedableRng> MockChain<R> {
+        pub fn new(mut rng: R) -> Result<Self, rand::Error> {
+            let account_rng = R::from_rng(&mut rng).expect("rng seeding failed");
+            let account_id_builder = AccountIdBuilder::new(account_rng);
+            Ok(Self {
+                chain: ChainMmr::default(),
+                data: DefaultMerkleStore::default(),
+                blocks: vec![],
+                nullifier_db: TieredSmt::default(),
+                rng,
+                account_id_builder,
+                objects: Objects::new(),
+                pending_objects: Objects::new(),
+            })
+        }
+
+        /// Creates the next block.
+        ///
+        /// This will also make all the objects currently pending available for use.
+        pub fn seal_block(&mut self) -> BlockHeader {
+            // Update the DBs
+            // TODO: self.nullifier
+            // TODO: accounts
+            // TODO: notes
+
+            let previous = self.blocks.last();
+            let block_num: u64 = self.blocks.len().try_into().expect("usize to u64 failed");
+            let block_num: Felt = block_num.into();
+            let peaks = self.chain.mmr().accumulator();
+            let chain_root: Digest = peaks.hash_peaks().into();
+            let prev_hash = previous.map_or(Digest::default(), |header| header.hash());
+            let nullifier_root = self.nullifier_db.root();
+            let version = Felt::ZERO;
+            let timestamp =
+                previous.map_or(TIMESTAMP_START, |header| header.timestamp() + TIMESTAMP_STEP);
+
+            // TODO: Set batch_root and proof_hash to the correct values once the kernel is
+            // available.
+            let batch_root = Digest::default();
+            let proof_hash = Digest::default();
+
+            let new = BlockHeader::new(
+                prev_hash,
+                block_num,
+                chain_root,
+                Default::default(),
+                nullifier_root,
+                Default::default(),
+                batch_root,
+                proof_hash,
+                version,
+                timestamp,
+            );
+
+            self.blocks.push(new);
+            self.chain.mmr_mut().add(new.hash());
+            self.objects.extend(&mut self.pending_objects);
```

Comment:
I haven't added any validation code because all objects are created by the builders, which should never generate conflicting IDs because of the RNG.